src/soap/handlers/GetSitesObject.js

```diff
@@ -48,16 +64,30 @@ export async function* getSitesObject(
         // TODO: Sort! We should default to _id. Check other services.
         $limit: 2000
       },
-      request.params && request.params.org
-        ? { organization_id: await helpers.orgId(request.params.org) }
+      organization &&
+        organization.data &&
+        organization.data.length &&
+        organization.data[0]._id
+        ? { organization_id: await helpers.orgId(organization.data[0]._id) }
         : undefined,
       sites.length
-        ? { _id: { $in: sites.map(str => str.split(':')[1]) } }
+        ? { slug: { $in: sites.map(str => str.split(':')[1]) } }
```

Comment:
There are two nuances with this.

In the current WOF service, you pass `serviceCode:SiteCode` and if it's an invalid serviceCode, nothing is returned. The current code doesn't emulate this. It basically ignores what is on the left side of the colon.

It's kinda redundant too, the org is in the route (it needs to be), and the serviceCode is the SAME as the org slug. So it's like saying the same thing twice.

The other nuance is the siteCode is mapped to a portion of the station slug. Since station slugs have to be globally unique in the system, Collin contrived it by concatenating the org slug and station slugs.

Let me spec out the precise mechanism here and then you can code it. I know how it should probably work with the combination of org in the route and the SOAP params. We were rushing Friday and I've had time to think about this. It pretty close. I'll write up a spec Monday morning.

Reply:
Changed 
```
sites.length
        ? {
            slug: {
              $in: sites.map(str => {
                const parts = str.split(':')
                return helpers.safeName(
                  (request.params.org || parts[0] || '-') +
                    '-' +
                    (parts[1] || '-')
                )
              })
            }
          }
        : undefined
```